### ISSUE TITLE
list of possible functions to be included in new data cleaning package in `easystats`

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/easystats#97]
Please edit the list as you come across such functions across core packages:

- [ ] `effectsize::adjust`

**Comment by DominiqueMakowski** [comment]
Data is the food that models (monsters) eat, so data preparation is kind of like cleaning and cooking the food. So for the logo, I'd see in the background a roundy monster with a napking tucked into its collar, looking forward his delicious meal (with its mouth dripping). In front, a delicious dish that is being peppered (i.e., data being prepared).

edit: something like this but from the front

![image](https://user-images.githubusercontent.com/8875533/104834752-7db51180-58dc-11eb-96b9-d1d70dc0828b.png)

**Comment by IndrajeetPatil** [comment]
Note to self: 
If any of these function were exported, continue to re-export them from respective packages for backward compatibility

**Comment by IndrajeetPatil** [comment]
Resolve this discussion before making these changes: https://github.com/easystats/effectsize/issues/92

**Comment by IndrajeetPatil** [comment]
What name did we land on? I like `dataperfect`, but I am not wedded to it.

https://github.com/easystats/easystats/issues/47#issuecomment-692776335

**Comment by bwiernik** [comment]
The reliability analysis and factor analysis functions seem rather odd and unlike the rest of the functions here. Seems like those scale/measure development functions ought to occupy their own space, particularly if eventually functions for CFA and related models were added. 

I like "dataprep" as a name.

**Comment by DominiqueMakowski** [comment]
Let's list all possibilities, and then we each sort them by order of preference, and then we compute some sort of score 🙃 

Dom's order:
- datatools
- dataprep
- datacook
- dataclean
- dataperfect
- easydata
- datamanage
- datamanip
- dataready

**Comment by DominiqueMakowski** [comment]
> The reliability analysis and factor analysis functions seem rather odd and unlike the rest of the functions here. Seems like those scale/measure development functions ought to occupy their own space, particularly if eventually functions for CFA and related models were added.

Yeah but then I don't want us to fall into the trap our creating our own tentacled universe with a big list of narrowly scoped packages that users would be lost into. It would confuse them and deter developers for which our "light dependencies" philosophy is a major selling point (so far we managed to explain that the only dependencies we have are "our own packages" so "it doesn't really count", but still if it multiplies then people will raise eyebrows - and it will become a legit issue).

Moreover one of the core tenet of easystats' is that all the packages are somewhat related to one another in a structured hierarchy, culminating in the "easystats" verse. They are not meant as stand-alone (of course one can use only a couple - and it's what we do as developers - but it's not how they were ontologically thought of). I would even say that the reason why multiple packages exist in the first place, instead of just one big "easystats" package, is more practical aspects like maintainability, organization, structure, and development ease, than anything else. 

Finally, I doubt it is even possible to create a perfect and balanced discretization of feature / use case space, and so there will always be some functions that borderline fit the scope of their package, but to be fair this kind of stuff is likely to bother more us as developers with a tendency for neat and meaningful categorizations, than users that don't really care about whether the name of the package they load fits a function (I've loaded `car` 🚗 to run anovas for years before the thought even occurred to me that it could be an acronym 😅)

**Comment by mattansb** [comment]
My top picks:
- datatools - good, broad scope (:
- datacook reminiscent of the `recipe` package - but is that a bad thing?
- ~~dataprep~~ too close to the `prepdat` package...

**Comment by DominiqueMakowski** [comment]
datawizard 🧙 is suprisingly not taken it seems, kinda like datatools but with a more whimsical feel to it ☺️

**Comment by DominiqueMakowski** [comment]
how to clean the data: 

![](https://media.giphy.com/media/FHEjBpiqMwSuA/giphy.gif) 

How to tidy up and store data:

![](https://media.giphy.com/media/3BKf0I2PVxAfC/giphy.gif)

**Comment by mattansb** [comment]
datawizard - logo is a monster with a pointy hat, like Mickey Mouse in fantasia!

**Comment by IndrajeetPatil** [comment]
This is on GitHub, not on CRAN. So we will have to negotiate them if they are wedded to the idea and if it's okay with them if we take that name.

But if you think `datawizard` is a better name, then we don't have to discuss this with them.

**Comment by DominiqueMakowski** [comment]
I do prefer datawizard now, it's more catchy I think! and it avoids negotiation :p what do y'all think?

**Comment by mattansb** [comment]
Same (:

**Comment by strengejacke** [comment]
yeah, sounds good!

**Comment by DominiqueMakowski** [comment]
![](https://media.giphy.com/media/Qs75BqLW44RrP0x6qL/giphy.gif)

**Comment by IndrajeetPatil** [comment]
I have laid out the bones, will start filling in flesh this weekend:
https://github.com/easystats/datawizard

Feel free to make changes you want until then.
